templates/slot/components/upgradescheduler.html

```diff
@@ -0,0 +1,161 @@
+{{ define "upgrade_scheduler" }}
+  <style>
+    .upgrade-scheduler #datepicker {
+      width: auto;
+    }
+    .upgrade-scheduler .table-container {
+      max-width: 100%;
+      overflow-x: auto;
+      max-height: 500px;
+    }
+    .upgrade-scheduler table {
+      position: relative;
+    }
+    .upgrade-scheduler thead {
+      position: sticky;
+      top: 0;
+    }
+    .upgrade-scheduler thead {
+      background-color: var(--bg-color);
+      box-shadow: 0 2px 2px -1px var(--shadow-light);
+    }
+    .upgrade-scheduler th,
+    .upgrade-scheduler td {
+      padding: 4px;
+    }
+
+    .upgrade-scheduler td:first-child,
+    .upgrade-scheduler td:nth-child(2) {
+      border-bottom: solid 1px var(--shadow-light);
+    }
+
+    .upgrade-scheduler td.perfect {
+      background-color: #66bb6a;
+    }
+
+    .upgrade-scheduler td.good {
+      background-color: #c8e6c9;
+    }
+
+    .upgrade-scheduler td.ok {
+      background-color: #fff9c4;
+    }
+
+    .upgrade-scheduler td.bad {
+      background-color: #ffcdd2;
+    }
+  </style>
+  <div class="m-2 d-flex flex-column upgrade-scheduler">
+    <div class="m-2 mb-4">
+      <label>Upgrade Date</label>
+      <input id="datepicker" type="text" class="form-control" />
+    </div>
+    <div class="table-container">
+      <table>
+        <thead>
+          <tr>
+            <th class="px-2">Epoch</th>
+            <th class="px-2">Start Slot</th>
+            <th class="px-2">UTC</th>
+            <th class="px-2">Moscow</th>
+            <th class="px-2">Los Angeles</th>
+            <th class="px-2">New York</th>
+            <th class="px-2">Brisbane</th>
+            <th class="px-2">Local Time</th>
+          </tr>
+        </thead>
+        <tbody id="upgrade-table"></tbody>
+      </table>
+    </div>
+  </div>
+  <small class="d-flex justify-content-end align-items-center m-2">Inpired by <a href="https://slots.symphonious.net" class="ml-2">https://slots.symphonious.net</a></small>
+  <script>
+    var genesisTimeStamp = {{ .ChainGenesisTimestamp }}
+    var secondsPerSlot = {{ .ChainSecondsPerSlot }}
+    var slotsPerEpoch = {{ .ChainSlotsPerEpoch }}
+    const timeZones = ["UTC", "Europe/Moscow", "America/Los_Angeles", "America/New_York", "Australia/Brisbane", undefined] //undefined for local time zone
+
+    function getClassForNumber(num) {
+      if (num % 100 === 0) {
+        return "perfect"
+      } else if (num % 10 === 0) {
+        return "good"
+      }
+    }
+
+    function getTimeClass(date, tz) {
+      const tzDate = date.setZone(tz, { keepLocalTime: true })
+      const hour = tzDate.plus({ minutes: tzDate.o }).hour
+      if (hour >= 8 && hour <= 16) {
+        return "perfect"
+      } else if (hour >= 7 && hour <= 20) {
+        return "good"
+      } else if (hour >= 5 && hour <= 23) {
+        return "ok"
+      }
+      return "bad"
+    }
```

Comment:
The color scheme implemented matches the one used in https://slots.symphonious.net/ but does NOT match the according BIDS' description. Is that intended?

Reply:
Yes, discussed it with Inan ... will add a comment in the BIDS